app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/campaign/view/CampaignListFragment.kt

```diff
@@ -67,41 +62,70 @@ class CampaignListFragment : Fragment(), CampaignListContract.View {
             //                        openCourseDetail(it)
         }
 
+        initializeRecyclerView()
+
+        sharedPrefs.cookies?.let {(activeCampaignViewModel.fetchCampaignList(it)) }
+        setData()
+        showProgressBar()
+        showMessage()
+
+    }
+
+
+    /**
+     *   This initializes the recyclerview
+     */
+
+    fun initializeRecyclerView(){
         recyclerCampaignList?.apply {
             layoutManager = LinearLayoutManager(context)
             setHasFixedSize(true)
             adapter = campaignListRecyclerAdapter
         }
-
-        sharedPrefs.cookies?.let { campaignListPresenter.requestCampaignList(it) }
     }
 
-    override fun setData(data: ExploreCampaignsResponse) {
-        Timber.d(data.toString())
-        if (data.campaigns.isNotEmpty()) {
-            campaignList = data.campaigns
-            recyclerCampaignList?.visibility = View.VISIBLE
-            campaignListRecyclerAdapter.setData(data.campaigns)
-            campaignListRecyclerAdapter.notifyDataSetChanged()
-            textViewNoCampaigns?.visibility = View.GONE
-        } else {
-            recyclerCampaignList?.visibility = View.GONE
-            textViewNoCampaigns?.visibility = View.VISIBLE
-        }
+    /**
+     *   This sets the data to be displayed on the recyclerview based on available data
+     */
+    fun setData(){
+        activeCampaignViewModel.data.observe(this, androidx.lifecycle.Observer {
+            Timber.d( "hello",it.toString())
+            if (it.isNotEmpty()) {
+                recyclerCampaignList?.visibility = View.VISIBLE
+                campaignListRecyclerAdapter.setData(it)
+                campaignListRecyclerAdapter.notifyDataSetChanged()
+                textViewNoCampaigns?.visibility = View.GONE
+
+            } else {
+                recyclerCampaignList?.visibility = View.GONE
+                textViewNoCampaigns?.visibility = View.VISIBLE
+            }
+
+        })
     }
 
-    override fun showProgressBar(show: Boolean) {
-        progressBar?.visibility = if (show) {
-            View.VISIBLE
-        } else {
-            View.GONE
-        }
+    /**
+     *   This shows the progressbar
+     */
+    fun showProgressBar(){
+        activeCampaignViewModel.progressbar.observe(this, androidx.lifecycle.Observer {
+            if(it == true) progressBar.visibility = View.VISIBLE else progressBar.visibility = View.GONE
```

Comment:
`progressBar.visibility = if (it) View.VISIBLE else View.GONE`
Using `kt` you can use the if as a return statement